controllers/replicaset_controller_test.go

```diff
@@ -173,7 +173,103 @@ func TestStatefulSet_IsCorrectlyConfigured(t *testing.T) {
 	assert.NoError(t, err)
 	assert.NotNil(t, acVolume.Secret, "automation config should be stored in a secret!")
 	assert.Nil(t, acVolume.ConfigMap, "automation config should be stored in a secret, not a config map!")
+}
 
+func TestGuessEnterprise(t *testing.T) {
+	type testConfig struct {
+		setArgs            func(t *testing.T)
+		mdb                mdbv1.MongoDBCommunity
+		expectedEnterprise bool
+	}
+	tests := map[string]testConfig{
+		"No override and Community image": {
+			setArgs: func(t *testing.T) {
+				t.Setenv(construct.MongodbRepoUrl, "docker.io/mongodb")
+				t.Setenv(construct.MongodbImageEnv, "mongodb-community-server")
+			},
+			mdb:                mdbv1.MongoDBCommunity{},
+			expectedEnterprise: false,
+		},
+		"No override and Enterprise image": {
+			setArgs: func(t *testing.T) {
+				t.Setenv(construct.MongodbRepoUrl, "docker.io/mongodb")
+				t.Setenv(construct.MongodbImageEnv, "mongodb-enterprise-server")
+			},
+			mdb:                mdbv1.MongoDBCommunity{},
+			expectedEnterprise: true,
+		},
+		"Assuming enterprise manually": {
+			setArgs: func(t *testing.T) {
+				t.Setenv(construct.MongodbRepoUrl, "docker.io/mongodb")
+				t.Setenv(construct.MongodbImageEnv, "mongodb-community-server")
+				t.Setenv(construct.MongoDBAssumeEnterpriseEnv, "true")
+			},
+			mdb:                mdbv1.MongoDBCommunity{},
+			expectedEnterprise: true,
+		},
+		"Assuming community manually": {
+			setArgs: func(t *testing.T) {
+				t.Setenv(construct.MongodbRepoUrl, "docker.io/mongodb")
+				t.Setenv(construct.MongodbImageEnv, "mongodb-enterprise-server")
+				t.Setenv(construct.MongoDBAssumeEnterpriseEnv, "false")
+			},
+			mdb:                mdbv1.MongoDBCommunity{},
+			expectedEnterprise: false,
+		},
+		"Enterprise with different repo": {
+			setArgs: func(t *testing.T) {
+				t.Setenv(construct.MongodbRepoUrl, "some_other_repo.com/some_other_org")
+				t.Setenv(construct.MongodbImageEnv, "mongodb-enterprise-server")
+			},
+			mdb:                mdbv1.MongoDBCommunity{},
+			expectedEnterprise: true,
+		},
+		"Community with different repo": {
+			setArgs: func(t *testing.T) {
+				t.Setenv(construct.MongodbRepoUrl, "some_other_repo.com/some_other_org")
+				t.Setenv(construct.MongodbImageEnv, "mongodb-community-server")
+			},
+			mdb:                mdbv1.MongoDBCommunity{},
+			expectedEnterprise: false,
+		},
+		// This one is a corner case. We don't expect users to fall here very often as there are
+		// dedicated variables to control this type of behavior.
+		"Enterprise with StatefulSet override": {
```

Comment:
can we add a test case for the other way around? Setting env var to enterprise, but sts wrapper to non enterprise

Reply:
This is great spot! Indeed, there's a bug in the implementation but I also had to add an explicit check if the overridden image is really there. Please re-check the code.